test/spmd/tensor/test_pointwise_ops.py

```diff
@@ -8,52 +11,184 @@
 )
 from spmd.tensor.dispatch import OpSchema
 
+from spmd.tensor import distribute_tensor
 from spmd.tensor.ops.pointwise_ops import pointwise_rule
 from spmd import DeviceMesh, DTensor
-from spmd.tensor.placement_types import Shard, Replicate, _Partial, DTensorSpec
+from spmd.tensor.placement_types import (
+    Shard,
+    Replicate,
+    _Partial,
+    DTensorSpec,
+    Placement,
+)
 from torch.distributed.distributed_c10d import ReduceOp
 from torch.testing._internal.common_distributed import skip_if_lt_x_gpu
 
 
+def deepcopy_convert_to_dtensor(
+    val: Any,
+    device_mesh: DeviceMesh,
+    placements: Sequence[Placement],
+) -> Any:
+    """
+    Recursively convert (over Sequence and Dict types) Tensors into DTensors.
+
+    :param device_mesh: the DeviceMesh to use.
+    :param placements: the Placement list to use.
+    :return: the transformed structure.
+    """
+    if isinstance(val, Tensor):
+        return distribute_tensor(
+            val,
+            device_mesh=device_mesh,
+            placements=placements,
+        )
+
+    if isinstance(val, Sequence):
+        return [
+            deepcopy_convert_to_dtensor(
+                x,
+                device_mesh=device_mesh,
+                placements=placements,
+            )
+            for x in val
+        ]
+
+    if isinstance(val, Dict):
+        return {
+            k: deepcopy_convert_to_dtensor(
+                v,
+                device_mesh=device_mesh,
+                placements=placements,
+            )
+            for k, v in val.items()
+        }
+
+    return val
+
+
+def deepcopy_convert_from_dtensor(val: Any) -> Any:
+    """
+    Recursive convert any DTensor to local Tensor.
+
+    :param val: the structure to coerce.
+    :return: the coerced structure.
+    """
+    if isinstance(val, DTensor):
+        return val.to_local()
+
+    if isinstance(val, Sequence):
+        return [deepcopy_convert_from_dtensor(x) for x in val]
+
+    if isinstance(val, Dict):
+        return {k: deepcopy_convert_from_dtensor(v) for k, v in val.items()}
+
+    return val
+
+
 class DistElementwiseOpsTest(DistTensorTestBase):
+    def _compare_pairwise_ops(
+        self,
+        *,
+        device_mesh: DeviceMesh,
+        placements: Sequence[Placement],
+        op: Callable,
+        reset_seed: Optional[Callable] = None,
+        args: Sequence[Any] = tuple(),
+        kwargs: Optional[Dict[str, Any]] = None,
+    ):
+        if not kwargs:
+            kwargs = {}
+
+        torch.manual_seed(self.rank)
+
+        dargs = deepcopy_convert_to_dtensor(
```

Comment:
would pytree help here? I think you can do 

  import torch._pytree as pytree

  flattened_args = pytree.tree_flatten(args)
  dtensor_args  = [convert_to_dtensor(x) for x in flattened_args]
  d_args = pytree.tree_unflatten(flattened_args)

or :

   d_args = pytree.tree_map(args, convert_to_dtensro)

Reply:
done